src/core/raster/qgsrastershader.cpp

```diff
@@ -169,6 +170,7 @@ void QgsRasterShader::readXml( const QDomElement& elem )
   if ( !colorRampShaderElem.isNull() )
   {
     QgsColorRampShader* colorRampShader = new QgsColorRampShader();
+    colorRampShader->setColorRampName( colorRampShaderElem.attribute( "colorRampName", "" ) );
```

Comment:
As mentioned earlier, it's more efficient to avoid `""`" string constructors and just use `QString()` instead. Ie:

    colorRampShader->setColorRampName( colorRampShaderElem.attribute( "colorRampName", QString() ) );